src/Models/Market.php

```diff
@@ -52,6 +59,21 @@ protected static function boot()
             }
         });
 
+        static::updating(function ($market) {
+            $other_market = Market::where('slug', $market->slug)->where('id', "<>", $market->id)->first();
+
+            if ($other_market) {
+                return false;
+            }
+
+            $market->updatingSlug();
+
+            /*@todo this approach is better, the relations are not working*/
+            $market->page()->slug = $market->slug;
+            $market->page()->title = $market->title;
+            $market->page()->save();
```

Comment:
Why are these fields duplicated on the market and the page?  If the same value is always required, whats the benefit of including them in the market model too?

Reply:
I had to add slug back to the Market & Location so that we have a unique identifier for them that won't be changed.

The Title is the name of the Market and will be used in different places than the Page title. Page Title will later become the HTML frontend display title. For now, we'll create it the same, but it can be updated on the Page resource to be something different. Slug won't be able to be updated on the Page and will only be able to be updated from Market or Location.

Reply:
But, the page slug is the only slug that matters for routing now - and it already has a unique requirement.  Including the duplicated fields on market and location is really just creating opportunities for things to get out of sync in the 2 models causing breakage.  But, if you keep them, you'll need to add the event listener for Page model changes I described in another comment.

Reply:
Thank you. I like this approach, so the Page is created with the same Slug when the Market is created. I agree it adds a little complexity, but I think it is worth it. Let's add the following:

- Market & Location saving boot method will update Page slug
- On the Page model, if it has a Market or Location relationship, then the Slug can't be updated in Nova

Reply:
@drewroberts  - can you create an issue in the blog package to make sure this part isn't forgotten - "On the Page model, if it has a Market or Location relationship, then the Slug can't be updated in Nova"

Reply:
Done.

https://github.com/drewroberts/blog/issues/65